app/src/main/java/com/example/vybe/CreateAccountActivity.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    // Remove the listener on stop as we don't want to redirect? not sure
+    @Override
+    protected void onStop() {
+        super.onStop();
+        if (mAuthListener != null) {
+            mAuth.removeAuthStateListener(mAuthListener);
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_create_account);
+        usernameField = findViewById(R.id.username_create);
+        emailField = findViewById(R.id.email_create);
+        passwordField = findViewById(R.id.password_create);
+        confirmBtn = findViewById(R.id.confirm_button);
+
+        user = new User();
+
+        mAuth = FirebaseAuth.getInstance();
+        db = FirebaseFirestore.getInstance();
+
+        mAuthListener = new FirebaseAuth.AuthStateListener() {
+            @Override
+            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+                FirebaseUser userFB = firebaseAuth.getCurrentUser();
+                // User is logged in
+                if (userFB != null) {
+                    Intent intent = new Intent(CreateAccountActivity.this, MyVibesActivity.class);
+                    startActivity(intent);
+                }
+            }
+        };
+
+        confirmBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // Check all the fields are filled in for sign in
+                if (!isEmpty(usernameField) && !isEmpty(emailField) && !isEmpty(passwordField)) {
+                    user.setUsername(getTrimmedString(usernameField));
+                    // Check Firestore if username is already taken
+                    db.collection("Users").whereEqualTo("username", user.getUsername()).get()
+                        .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                            @Override
+                            public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                                verifyCreateAccount(task);
+                            }
+                        });
+                }
+
+            }
+        });
+    }
+
+    /**
+     * Helper function to check if an edit text field is empty
+     *
+     * @param et EditText that we want to check
+     * @return Returns true if the field is empty, false it is filled
+     */
+    public static boolean isEmpty(EditText et) {
+        return et.getText().toString().trim().length() == 0;
+    }
+
+    /**
+     * Helper function to get trimmed string from edit text field
+     *
+     * @param et EditText that we want to get the string from
+     * @return Returns the string from the EditText trimmed
+     */
+    public static String getTrimmedString(EditText et) {
+        return et.getText().toString().trim();
+    }
+
+    public void verifyCreateAccount(Task<QuerySnapshot> task){
```

Comment:
This method is long and doing two things. I suggest splitting it up into a "checking if account is in database" method  and "create account"

Reply:
Done 😃